desktop/extensions-sdk/build/set-up/backend-extension-tutorial.md

```diff
@@ -0,0 +1,419 @@
+---
+
+Your extension can ship a backend part with which the frontend can interact with. This page provides information on
+why and how to add a backend.
+
+> Note
+>
+> Before you start, make sure you have installed the latest version of [Docker Desktop](https://www.docker.com/products/docker-desktop/).
+
+> Note
+>
+> If you want to start a codebase for your new extension, our [Quickstart guide](../../quickstart.md) and `docker extension init <my-extension>` provides a better base for your extension as it is more up-to-date and related to your install of Docker Desktop.
+
+## Why add a backend?
+
+Thanks to the Docker Extensions SDK, most of the time you should be able to do what you need from the Docker CLI
+directly from [the frontend](./frontend-extension-tutorial.md).
+
+Nonetheless, there are some cases where you might need to add a backend to your extension. So far, extension
+builders have used the backend to:
+- Store data in a local database and serve them back with a REST API.
+- Store the extension state, like when a button starts a long-running process, so that if you navigate away
+  from the extension and come back, the frontend can pick up where it left off.
+
+## Add a backend to the extension
+
+If you created your extension using the `docker extension init` command, you already have a backend set up. If it is
+not the case, then you have to first create a `vm` directory that will contain the code and update the Dockerfile to
+containerize it.
+
+Here is the extension folder structure with a backend:
+
+```bash
+.
+├── Dockerfile # (1)
+├── Makefile
+├── metadata.json
+├── ui
+    └── index.html
+└── vm # (2)
+    ├── go.mod
+    └── main.go
+```
+
+1. Contains everything required to build the backend and copy it in the extension's container filesystem.
+2. The source folder that contains the backend code of the extension
+
+Although you can start from an empty directory or from the `vm-ui extension` [sample](https://github.com/docker/extensions-sdk/tree/main/samples){:target="_blank" rel="noopener" class="_"},
+it is highly recommended that you start from the `docker extension init` command and change it to suit your needs.
+
+> **Tip**
+>
+> The `docker extension init` generates a Go backend. But you can still use it as a starting point for
+> your own extension and use any other language like Node.js, Python, Java, .Net, or any other language and framework.
+{: .tip }
+
+On this tutorial, the backend service simply exposes one route that returns a JSON payload that says "Hello".
+
+```json
+{ "Message": "Hello" }
+```
+
+> **Important**
+>
+> We recommend that, the frontend and the backend communicate through sockets (and named pipes on Windows) instead of
+> HTTP. On one hand, because it will prevent port collision with any other running application or container running
+> on the host. On the other hand, because some Docker Desktop users might be unable to open ports on their machines.
```

Comment:
```suggestion
> on the host. On the other hand, because some Docker Desktop users are running in constrained environments where they can't open ports on their machines.
```